File: git/457821_final/home-organization-app/backend/app/schemas/daily_focus.py
from __future__ import annotations

from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class DailyFocusTaskMini(BaseModel):
    id: int
    title: str
    room_id: Optional[int] = None
    due_date: Optional[datetime] = None
    completed: bool

    model_config = ConfigDict(from_attributes=True)


class DailyFocusRead(BaseModel):
    id: Optional[int] = None
    user_id: int
    date: date
    task_id: Optional[int] = None
    completed_at: Optional[datetime] = None
    task: Optional[DailyFocusTaskMini] = None

    model_config = ConfigDict(from_attributes=True)


class DailyFocusCompleteIn(BaseModel):
    task_id: Optional[int] = None


class DailyFocusRefreshIn(BaseModel):
    preferred_room_id: Optional[int] = None

File: git/457821_final/home-organization-app/frontend/src/schemas/daily_focus.ts
/**
 * Daily Focus schemas
 * Matches backend app/schemas/daily_focus.py
 */
export interface DailyFocusTaskMini {
  id: number;
  title: string;
  room_id: number | null;
  due_date: string | null; // ISO datetime string
  completed: boolean;
}

export interface DailyFocusRead {
  id: number | null;
  user_id: number;
  date: string; // ISO date string (YYYY-MM-DD)
  task_id: number | null;
  completed_at: string | null; // ISO datetime string
  task: DailyFocusTaskMini | null;
}

export interface DailyFocusCompleteIn {
  task_id?: number | null;
}

export interface DailyFocusRefreshIn {
  preferred_room_id?: number | null;
}
